pkg/services/sqlstore/component_interfaces.go

```diff
@@ -102,37 +125,43 @@ func (s storeDS) Update(ctx context.Context, ds datasource.CR) error {
 	return s.ss.UpdateDataSourceByUID(ctx, cmd)
 }
 
-func (s storeDS) Delete(ctx context.Context, uid string) error {
+func (s storeDS) Delete(ctx context.Context, name types.NamespacedName) error {
 	return s.ss.DeleteDataSource(ctx, &models.DeleteDataSourceCommand{
-		UID:   uid,
+		Name:  name.Name,
 		OrgID: 1, // hardcode for now, TODO
 	})
 }
 
 // oldToNew doesn't need to be method, but keeps things bundled
-func (s storeDS) oldToNew(ds *models.DataSource) datasource.CR {
-	jdMap := ds.JsonData.MustMap()
-	cr := datasource.CR{
-		Spec: datasource.Model{
-			Type:              ds.Type,
-			Access:            string(ds.Access),
-			Url:               ds.Url,
-			Password:          ds.Password,
-			Database:          ds.Database,
-			User:              ds.User,
-			BasicAuth:         ds.BasicAuth,
-			BasicAuthUser:     ds.BasicAuthUser,
-			BasicAuthPassword: ds.BasicAuthPassword,
-			WithCredentials:   ds.WithCredentials,
-			IsDefault:         ds.IsDefault,
-			JsonData:          jdMap,
-			// SecureJsonData: TODO,
-			//Version: ds.Version,
-			// Note: Not mapped is created / updated time stamps
-		},
+func (s storeDS) oldToNew(ds *models.DataSource, result runtime.Object) error {
```

Comment:
Really looking forward to getting this in particular into the component, whew

Reply:
Yeah, that'd be the next step to figure out - it's related to generalising the Store somehow. Right now it's generalised on the level of interfaces, but the actual dispatching is hard-coded using Wire's DI mechanism.

I would like to try & implement some way of dynamically registering / dispatching store implementations, similar to Kubernetes' clientset.